routes/profile.js

```diff
@@ -0,0 +1,57 @@
+const http = require("http");
+const express = require("express")
+const cookieParser = require("cookie-parser")
+const bodyParser = require("body-parser")
+const ObjectId = require('mongoose').Types.ObjectId;
+const { Schema,
+    userSchema,
+    User,
+    connect,
+    sessionSchema,
+    Session,
+    postSchema,
+    Post,
+    replySchema,
+    Reply } = require("../database.js");
+
+const router = express.Router();
+
+router.get("/:name", async (req, res) => {
+    try {
+        const pageUser = await User.find({ username: req.params.name });
+        if (!pageUser[0]) {
+            res.render('profile', { user: { username: "this user does not exist" } });
+            return;
+        }
+        const allPostOfUserQ = await Post.find({
+            user:
+            {
+                id: pageUser[0].id,
+                name: pageUser[0].username
+            }
+        })
```

Comment:
You don't need to search by both user ID and user name, the ID is a globally unique identifier, it's enough to search by it.

Reply:
The ID will not repeat between users with the same name, the name might, that's the point about an ID, it never repeats and never changes, even if you delete the user, that ID will never return.

Also, one of the cool parts about Mongo is that you are able to query deep into the object with no problems. With pure mongo you are able to issue the query `{ user: { id: 'whatever' } }` and it would work. Mongoose should allow you to do the same.